### PR TITLE
[BE-4789]: Initially fixed for wf step templates

### PR DESCRIPTION
**Pull request by boztopuz**

### CODE REVIEW

**Comment by ftmzhrsckldr** [comment]
> @ftmzhrsckldr I updated all WF step docs to have the same template.The changes were applied to each step so that the templates of all step documents are the same. Please check here before you start the review.
> 
> **Looking at these before you start will help us to spot any logical errors or mistakes early on and avoid wasting time. Because most of the changes are the same and repetitive.**
> 
> **What has changed?**
> 
> **1- All required Prequisites introduction sentences have been changed as follows.** `Before running the **<Step Name>** step, you must complete certain prerequisites, as detailed in the table below:`
> 
> **2- Corrected the formatting of all info, caution, danger etc boxes in the .md file.** **3- Added missing $ signs in all input tables.** **4- Removed $ sign in all output tables.** **5- Removed output section in docs for the steps that has not output variables.** **6- All sensitive variable danger boxes have been replaced with this.**
> 
> ```
> danger Sensitive Variables
> 
> Please do not use sensitive variables such as **Username**, **Password**, **API Key**, or **Personal Access Key** directly within the step.
> 
> We recommend using [**Environment Variables**](/environment-variables/managing-variables) groups for such sensitive variables.
> ```
> 
> **7- Duplicated post processor step doc has been deleted.** **8- All FAQ headings have been edited.** **9- Replaced the introductory sentences of all input headers with this.** `This step contains some input variable(s). It needs these variable(s) to work. The table below gives explanation for this variable(s).`
> 
> **10- Replaced the introductory sentences of all output headers with this one.** `The output(s) resulting from the operation of this component are as follows:`
> 
> **11- The introductory sentence under the prerequisites heading for steps without prerequisites has been replaced with the following.** `There are no prerequisites required before using the **<Step Name>** step.` **12- Added missing prerequisites introductory sentence.**
> 
> Note: I have not change anything in Saucectl doc. Because we already have new version of it in this [PR](https://github.com/appcircleio/appcircle-docusaurus/pull/645).

@boztopuz I did not see any problems with what was commented. However, after reviewing the documents, if one of the templates we used does not match that document, I can add feedback. For example, if the warn message you gave in step 6 does not match the variable that should be secret in the doc, I can add feedback. Other than that, I did not see any problems.

By the way, there were errors in document linking in old documents. Instead of internal linking, there were links starting with "http...". Were you able to check these as well?

**Comment by boztopuz** [comment]
> By the way, there were errors in document linking in old documents. Instead of internal linking, there were links starting with "http...". Were you able to check these as well?

Unfortunately, I did not check the links which include http...., I will try to check as soon as possible all of them today.

**Comment by ftmzhrsckldr** [comment]
@boztopuz 
> 8- All FAQ headings have been edited.

Is this what you mean by that?

 "## Custom Script FAQ => ## FAQ"

Is there anything else I should check other than that?

**Comment by boztopuz** [comment]
@ftmzhrsckldr 

> Corrected the formatting of all info, caution, danger etc boxes in the .md file.
Old version of document danger info etc boxes are like this in md files

```
:::danger
danger details
:::
```

I changed all of these like

```
:::danger

danger details

:::
```

The reason I do this is that some errors can occur in the docs when using non-space format. We have experienced this in the past


> Removed output section in docs for the steps that has not output variables.
When you look at the documents, you will see that I did not delete all of them, I deleted only the sections that do not have a table and a specific description.

> All FAQ headings have been edited.
Yes that is true, I changed the faq sections like 

## Custom Script FAQ => ## FAQ

**Comment by ftmzhrsckldr** [comment]
# Feedback Summary

@boztopuz The initial documents were very well-prepared and required minimal feedback. However, as we progressed, the amount of feedback increased. I believe having too many changes in a single PR might have reduced focus.

I structured my feedback so that you can directly apply the suggestions if you approve them. However, for some unchanged paragraphs, GitHub did not allow me to add feedback. In those cases, I made corrections by linking them to the nearest editable line.  

### Feedback Summary  

1. Many links were correctly updated, but some were missed in certain documents (e.g., links still starting with "http"). Since some untouched files (like `index.md`) contained these links, I fixed them with this commit: [143cb83](https://github.com/appcircleio/appcircle-docusaurus/pull/731/commits/143bc8303af12957b928f6bda274e96bae9c8922). Feel free to review it if needed.  
2. Some issues in untouched `index.md` files were not fixed since I could not leave feedback on unchanged lines. The following should be reviewed and corrected:  
   - Add `$` to environment variables where applicable.  
   - Separate `:::` blocks from content.  
3. A few points mentioned [here](https://github.com/appcircleio/appcircle-docusaurus/pull/731#issuecomment-2476067028) were missed in some documents:  
   - In earlier documents, reserved environment variables started with `$`, but this was inconsistent in later documents.  
   - Some `:::` blocks were not separated from content.  
   - The "Sensitive Variables" warning was missing in one document, though it was correct in others.  
4. **Metadata tags**: Many were incomplete or copied from other documents without proper editing.  
5. **Metadata descriptions**: Some contained unnecessary details (e.g., prerequisites).  
6. **Inconsistent formatting within the same document**:  
   - Some step names were bold, while others were not.  
   - File names were sometimes in backticks (``) but plain in other cases.  
7. Some typos were missed, and I have corrected them.  
8. A few documents did not follow the common pattern for source code links and descriptions. I fixed these.  
9. FAQ headings were corrected, but the content was not fully reviewed, leading to additional fe

CC: @csonuryilmaz

File: docs/workflows/common-workflow-steps/firebase-app-distribution.md [hunk 1]
```diff
@@ -86,10 +88,6 @@ Confidential information should be entered as a [secret environment variable](/e
 | `$AC_FIREBASE_GROUPS`          | Specifies the Firebase tester groups you want to invite.                                                                                               | Optional |
 | `$AC_FIREBASE_EXTRA_PARAMETERS` | Specifies extra command line parameters. Enter `--debug` for debug mode.                                                                                              | Optional |
 
-### Output Variables
-
-The **Firebase App Distribution** step generates no output variables. The step succeeds if the app is distributed successfully; otherwise, it fails.
```
[the remark this answers is not in the document]
Comment (boztopuz) [reply docs/workflows/common-workflow-steps/firebase-app-distribution.md:91]: Actually, I prefer to delete this section but I added last sentence as caution box you are right.

File: docs/workflows/flutter-specific-workflow-steps/firebase-deployment.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Deploy web applications effortlessly using Appcircle's Firebase Deployment compo
```
[the remark this answers is not in the document]
Comment (boztopuz) [reply docs/workflows/flutter-specific-workflow-steps/firebase-deployment.md:14]: This component is not related with mobile. Its web specific tool. The description of tool mentioning about the web app.

File: docs/workflows/ios-specific-workflow-steps/appdome-build-to-secure-for-ios.md [hunk 3]
```diff
@@ -54,14 +58,21 @@ We recommend using [Environment Variables](/environment-variables/) groups for s
...
-| `$AC_APPDOME_SECURED_IPA_PATH`         | Local path of the secured .ipa file. Available when 'Signing Method' set to `On-Appdome` or `Private-Signing` | Required |
-| `$AC_APPDOME_PRIVATE_SIGN_SCRIPT_PATH` | TLocal path of the .sh sign script file. Available when `Signing Method` set to `Auto-Dev-Signing`            | Required |
-| `$AC_APPDOME_CERTIFICATE_PATH`         | Local path of the Certified Secure Certificate `.pdf` file                                                    | Required |
+| `AC_APPDOME_SECURED_IPA_PATH`         | Local path of the secured .ipa file. Available when 'Signing Method' set to `On-Appdome` or `Private-Signing` | Required |
+| `AC_APPDOME_PRIVATE_SIGN_SCRIPT_PATH` | TLocal path of the .sh sign script file. Available when `Signing Method` set to `Auto-Dev-Signing`            | Required |
+| `AC_APPDOME_CERTIFICATE_PATH`         | Local path of the Certified Secure Certificate `.pdf` file                                                    | Required |
+
+---
+
+To access the source code of this component, please use the following link:
```
[the remark this answers is not in the document]
Comment (boztopuz) [reply docs/workflows/ios-specific-workflow-steps/appdome-build-to-secure-for-ios.md:77]: @ftmzhrsckldr  What is the change here?  I could not understand.

File: docs/workflows/ios-specific-workflow-steps/carthage.md [hunk 4]
```diff
@@ -14,27 +14,36 @@ import Screenshot from '@site/src/components/Screenshot';
 
 ### Prerequisites
 
-Appcircle will look for a [`Cartfile`](https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md) file in your repository and use it to install the dependencies. For this reason, **it should be used after the Git Clone step**.
+Before running the **Carthage** step, you must complete certain prerequisites, as detailed in the table below:
+
```
[the remark this answers is not in the document]
Comment (boztopuz) [reply docs/workflows/ios-specific-workflow-steps/carthage.md:18]: @ftmzhrsckldr I could not find duplicate white line unfortunately.

File: docs/workflows/ios-specific-workflow-steps/appdome-build-to-secure-for-ios.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (boztopuz) [reply docs/workflows/ios-specific-workflow-steps/appdome-build-to-secure-for-ios.md:77]: I found it :)

File: docs/workflows/ios-specific-workflow-steps/xcodebuild-for-devices.md [hunk 5]
```diff
@@ -41,15 +41,15 @@ This step contains some input variable(s). It needs these variable(s) to work. T
...
 | `$AC_SCHEME`                                  | Specifies the project scheme for build. If you filled in **`Configuration => Build Scheme`**, this variable comes from [Configuration](/build/platform-build-guides/building-ios-applications#build-configuration).                                   | Required |
-| `$AC_ARCHIVE_FLAGS`                           | Specifies the extra xcodebuild flag. For example: -quiet                                                                                                                                                                                                 |          |
+| `$AC_ARCHIVE_FLAGS`                           | Specifies the extra xcodebuild flag. For example: `-quiet`                                                                                                                                                                                                 |          |
```
Comment (ftmzhrsckldr) [inline docs/workflows/ios-specific-workflow-steps/xcodebuild-for-devices.md:47]: I just saw that feedback:
```suggestion
| `$AC_ARCHIVE_FLAGS`                           | Specifies the extra xcodebuild flag. For example: `-quiet`.                                                                                                                                                                                                 | Optional |
```

File: docs/workflows/ios-specific-workflow-steps/xcodebuild-for-devices.md [hunk 6]
```diff
@@ -81,19 +81,19 @@ https://github.com/appcircleio/appcircle-ios-build-sign-component
...
-- Change the name as "Custom Xcodebuild for Devices" for this custom script.
+- Copy all code lines from the `main.rb` file and paste them into the new **Custom Script** that you just added in your workflow.
+- Change the name as **Custom Xcodebuild for Devices** for this custom script.
 - Change "Execute With" picker as **Ruby**.
 - In the Ruby code, you can add the required codes to the end of the `xcodebuild` command.
 
 :::caution
 
-Before running the script, some variables must be changed, and new variables must be added to the custom script.
+Before running the script, some variables must be changed, and new variables must be added to the **custom script**.
```
Comment (ftmzhrsckldr) [inline docs/workflows/ios-specific-workflow-steps/xcodebuild-for-devices.md:96]: If it is a step name, should be follow:

```suggestion
Before running the script, some variables must be changed, and new variables must be added to the **Custom Script**.
```

File: docs/workflows/android-specific-workflow-steps/appsweep-mobile-security-testing.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 ---
 title: AppSweep Mobile Security Testing
 description: Learn how to use the AppSweep Mobile Security Testing workflow step in Appcircle
-tags: [android, mobile, android, security, testing, app sweep]
+tags: [android, mobile, android, security, testing, appsweep]
```
Comment (ftmzhrsckldr) [inline docs/workflows/android-specific-workflow-steps/appsweep-mobile-security-testing.md:4]: Can you delete duplicated tag:
```suggestion
tags: [android, mobile, security, testing, appsweep]
```

File: docs/workflows/android-specific-workflow-steps/android-unit-tests.md [hunk 8]
```diff
@@ -10,44 +10,48 @@ import Screenshot from '@site/src/components/Screenshot';
...
 
 <Screenshot url='https://cdn.appcircle.io/docs/assets/android-workflow-components-android-unit-tests_1.png'/>
 
 :::danger
-If you wish to view the test results on Appcircle's Test Reports page, it is essential to use the [Test Reports](https://github.com/appcircleio/appcircle-test-report-component) step after the **Android Unit Tests**. Please check out this document for more information: [Generating Test Report](https://docs.appcircle.io/continuous-testing/android-testing/running-android-unit-tests#generating-test-report)
+
+If you wish to view the test results on Appcircle's Test Reports page, it is essential to use the [Test Reports for Android](/workflows/android-specific-workflow-steps/test-reports-for-android) step after the **Android Unit Tests**. Please check out this document for more information: [Generating Test Report](/continuous-testing/android-testing/running-android-unit-tests#generating-test-report)
```
Comment (ftmzhrsckldr) [inline docs/workflows/android-specific-workflow-steps/android-unit-tests.md:27]: I just saw that I didn't give you feedback to add a point:
```suggestion
If you wish to view the test results on Appcircle's Test Reports page, it is essential to use the [Test Reports for Android](/workflows/android-specific-workflow-steps/test-reports-for-android) step after the **Android Unit Tests**. Please check out this document for more information: [Generating Test Report](/continuous-testing/android-testing/running-android-unit-tests#generating-test-report).
```

File: docs/workflows/android-specific-workflow-steps/android-unit-tests.md [hunk 9]
```diff
@@ -10,44 +10,48 @@ import Screenshot from '@site/src/components/Screenshot';
...
+| `$AC_VARIANTS`       | This variable specifies the project variant to be built. This variable can also be set via the build [Configuration](/build/build-process-management/build-profile-configuration). In Android Studio, you can find the available variants for your project. For more information, please refer to this [Android document](https://developer.android.com/build/build-variants).               | Required |
 | `$AC_PROJECT_PATH`   | Specifies the project path. If your project that needs to be built is **not located** in the root directory where it was cloned from Git, you should provide the subpath as a relative path. The default value is: `./`                                                                                                                                                                                                | Optional |
```
Comment (ftmzhrsckldr) [inline docs/workflows/android-specific-workflow-steps/android-unit-tests.md:44]: I just saw that I didn't give you feedback to add a point:
```suggestion
| `$AC_PROJECT_PATH`   | Specifies the project path. If your project that needs to be built is **not located** in the root directory where it was cloned from Git, you should provide the subpath as a relative path. The default value is: `./`.                                                                                                                                                                                               | Optional |
```

File: docs/workflows/android-specific-workflow-steps/app-center-android-distribution.md [hunk 10]
```diff
@@ -58,7 +63,7 @@ We recommend using [Environment Variables](https://docs.appcircle.io/environment
...
-| `$AC_APPCENTER_RELEASE_NOTES_PATH` | Specifies the release note path. If you use the [**Publishing Release Notes**](https://docs.appcircle.io/workflows/common-workflow-steps/publish-release-notes) component before this step, `release-notes.txt` will be used as release notes. The default value is `AC_OUTPUT_DIR/release-notes.txt`.                | Optional |
+| `$AC_APPCENTER_RELEASE_NOTES_PATH` | Specifies the release note path. If you use the [**Publishing Release Notes**](/workflows/common-workflow-steps/publish-release-notes) component before this step, `release-notes.txt` will be used as release notes. The default value is `AC_OUTPUT_DIR/release-notes.txt`.                | Optional |
 | `$AC_APPCENTER_MAPPING_PATH`       | Specifies the path of the `mapping.txt` file. Example: `$AC_REPOSITORY_DIR/build/app/outputs/mapping/release/mapping.txt`                                                                                                                                                                                             | Optional |
```
Comment (ftmzhrsckldr) [inline docs/workflows/android-specific-workflow-steps/app-center-android-distribution.md:67]: I just saw that I didn't give you feedback to add a point:
```suggestion
| `$AC_APPCENTER_MAPPING_PATH`       | Specifies the path of the `mapping.txt` file. Example: `$AC_REPOSITORY_DIR/build/app/outputs/mapping/release/mapping.txt`.                                                                                                                                                                                            | Optional |
```

**Comment by ftmzhrsckldr** [comment]
The vast majority of feedback has been implemented correctly.

@boztopuz `3` of the feedbacks I gave are new problems that occur after the update. The remaining `5` are deficiencies that I noticed only now but did not notice in my previous feedback.

Note: One of my previous feedback I marked unresolved. FYI

**Review by ftmzhrsckldr — APPROVED** [review]

**Pull request merged**